src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/TypeInfos/NativeFormat/NativeFormatRuntimeNamedTypeInfo.cs

```diff
@@ -79,7 +79,27 @@ public sealed override bool IsByRefLike
                         continue;
                     if (guidStringArgumentHandle.HandleType != HandleType.ConstantStringValue)
                         continue;
-                    return new Guid(guidStringArgumentHandle.ToConstantStringValueHandle(_reader).GetString(_reader));
+
+                    static Guid ReadAndParseGuid(Handle handle, MetadataReader reader)
+                    {
+                        if (handle.IsNil)
+                        {
+                            // We don't really have a parameter, so just match the name of the 'Parse' parameter
+                            ArgumentNullException.Throw("utf8Text");
+                        }
+
+                        // 68 characters is the maximum length of a guid, when formatted as 'X'
+                        Span<byte> destination = stackalloc byte[68];
```

Comment:
I'm not entirely sure this change is correct, as I _think_ it's allowed to also have a `[Guid]` with any number of leading/trailing whitespaces (for some reason). If that's the case, I suppose we could have this as a fast path, and perhaps fallback to a `string` if the length exceeds this? And rework the logic to at least try to eg. avoid decoding stuff twice if we go to the fallback perhaps? Thoughts? 😅

Reply:
Right, this looks like a corner-case bug farm.

Why do you need to copy the UTF8 string in the first place? Can you parse it directly?

Reply:
...Yes I don't see why not, just didn't think of that as I was trying to more or less keep the original pattern for some reason. Will do that, thank you for the suggestion! 😄